fbgemm_gpu/include/fbgemm_gpu/utils/cuda_prelude.cuh

```diff
@@ -73,7 +77,7 @@ DEVICE_INLINE T shfl_xor(
     const T val,
     int laneMask,
     int width = kWarpSize,
-    unsigned shfl_sync_mask = kFullWarpMask) {
+    unsigned shfl_sync_mask = static_cast<unsigned>(kFullWarpMask)) {
```

Comment:
Do we need explicit cast here? And isn't unsigned == unsigned int, why did we define it as uint64_t above?

Reply:
This is done to avoid massive "implicit-cast" warning spam if built with USE_ROCM. **shfl_sync_mask** is not used in this case anyway (why - is another question). Regarding uint64_t - see the comment [above](https://github.com/pytorch/FBGEMM/pull/3266#discussion_r1818558455)